wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/probe/server/ExecServerLivenessProbeConfigFactory.java

```diff
@@ -29,7 +29,7 @@ public HttpProbeConfig get(String workspaceId, Server server)
     try {
       URL url = new URL(server.getUrl());
       return new HttpProbeConfig(
-          url.getPort() == -1 ? 80 : url.getPort(),
+          url.getPort() == -1 ? url.getDefaultPort() : url.getPort(),
```

Comment:
Maybe it would be better to check `default port` here and throw `InternalInfrastructureException`. Otherwise, as far as I understand `IllegalArgumentException` may be thrown later.

Reply:
getDefaultPort will return 80 or 443 in case of https/http. And since we know that this server is http(s) it is OK I believe.

Reply:
Then it's OK to me.